toolkit/docs/how_it_works/0_intro.md

```diff
@@ -2,23 +2,308 @@ How The Build System Works
+    process[process]:::process
+    decision{{decision}}:::decision
+    goodstate([good state]):::goodState
+    badstate[[bad state]]:::badState
+```
+<br/>
 
-The build process can be split into three components: tooling, package generation, and image generation.
+### High-level RPM Flow
+Mariner is an RPM based distro. A single package (or RPM) is built using a combination of sources and a spec file. A signature file is used to verify the sources' hashes. 
+
+```mermaid
+flowchart LR
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% nodes
+    spec[/Spec/]:::io
+    localSourceTar[/Local Sources if present/]:::io
+    sigFile[/Signature file/]:::io
+    remoteSourceTar[/Remote Source/]:::io
+    patches[/Patches/]:::io
+    pack[Pack SRPM]:::process
+    srpm[/SRPM/]:::io
+    buildRPM[Build RPM]:::process
+    rpm[/RPM/]:::io
+
+    %% node flow
+    spec --> pack
+    localSourceTar --> pack
+    remoteSourceTar --> pack
+    patches --> pack
+    sigFile --> pack
+    pack -->srpm 
+    srpm --> buildRPM
+    buildRPM --> rpm
+```
+</br>
+
+### High-level Build Flow
+ The build process can be split into three components: tooling, package generation, and image generation. When building, Makefile options can be used to build Mariner entirely from end to end or to download prebuilt artifacts.
+
+
+```mermaid
+flowchart LR
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% nodes
+    buildTC[Build toolchain from scratch]:::process
+    pullTC[Pull toolchain from remote]:::process
+    tcRPMS[/Toolchain RPMs/]:::io
+    buildRPMS[Build RPMs from scratch]:::process
+    pullRPMS[Pull RPMs from remote]:::process
+    rpms[/RPMs/]:::io
+    buildImage[Build Image from scratch]:::process
+    image[(Image: Container vhd vhdx iso)]:::io
+
+    %% node flow
+    buildTC --> tcRPMS
+    pullTC --> tcRPMS
+    tcRPMS --> buildRPMS
+    buildRPMS --> rpms
+    pullRPMS --> rpms
+    rpms --> buildImage
+    buildImage --> image
+
+```
+
+</br>
 
 ### Tools
 The tooling consists of a set of `Makefiles`, various go programs, a bootstrapping environment running in `Docker`, and a `chroot` environment to build in. These are discussed [here](1_initial_prep.md). The toolkit is able to re-build all of the tools from source if desired.
```

Comment:
Suggestion: judging from [the documentation](https://www.gnu.org/software/make/manual/make.html#Makefiles), most probably the preference is to use small caps for "makefile" but even the owners are inconsistent.
```suggestion
The tooling consists of a set of makefiles, various Go programs, a bootstrapping environment running in `Docker`, and a `chroot` environment to build in. These are discussed [here](1_initial_prep.md). The toolkit is able to re-build all of the tools from source if desired.
```